src/edu/csus/ecs/pc2/core/PacketHandler.java

```diff
@@ -1622,15 +1624,26 @@ private void confirmSubmission(Packet packet, ClientId fromId) {
 
         Clarification submittedClarification = (Clarification) PacketFactory.getObjectValue(packet, PacketFactory.CLARIFICATION);
         Clarification clarification = contest.acceptClarification(submittedClarification);
-
-        // Send to team
+           
+        IElementObject[] ultimateDestination = (IElementObject[]) PacketFactory.getObjectValue(packet, PacketFactory.DESTINATION);
+        // Send to judge rather than team
         Packet confirmPacket = PacketFactory.createClarSubmissionConfirm(contest.getClientId(), fromId, clarification);
         controller.sendToClient(confirmPacket);
 
         // Send to clients and other servers
         if (isServer()) {
             controller.sendToJudgesAndOthers(confirmPacket, true);
         }
+        //check if clarification is actually an announcement clarification.
+        if (clarification.getQuestion().equals("") && clarification.isAnswered()) {
+            if (ultimateDestination == null || ultimateDestination.length == 0) {
+                controller.sendToTeams(confirmPacket);//TODO modift to accomodate ultimate destination
```

Comment:
I don't understand what the "TODO" is for.

Reply:
removed TODO